apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -81,256 +95,183 @@ router.post(
+        );
+      }
+      let qid_new;
+      try {
+        qid_new = path.normalize(req.body.id);
+      } catch (err) {
+        throw new Error(`Invalid QID (could not be normalized): ${req.body.id}`);
+      }
+      if (res.locals.question.qid === qid_new) {
+        res.redirect(req.originalUrl);
+      } else {
+        const editor = new QuestionRenameEditor({
+          locals: res.locals,
+          qid_new: qid_new,
         });
-      });
-    }
-  } else if (req.body.__action === 'copy_question') {
-    debug('Copy question');
-    if (idsEqual(req.body.to_course_id, res.locals.course.id)) {
-      // In this case, we are making a duplicate of this question in the same course
-      const editor = new QuestionCopyEditor({
-        locals: res.locals,
-      });
-      editor.canEdit((err) => {
-        if (ERR(err, next)) return;
-        editor.doEdit((err, job_sequence_id) => {
-          if (ERR(err, (e) => logger.error('Error in doEdit()', e))) {
-            res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
-          } else {
-            debug(
-              `Get question_id from uuid=${editor.uuid} with course_id=${res.locals.course.id}`,
-            );
-            sqldb.queryOneRow(
-              sql.select_question_id_from_uuid,
-              { uuid: editor.uuid, course_id: res.locals.course.id },
-              (err, result) => {
-                if (ERR(err, next)) return;
-                flash(
-                  'success',
-                  'Question copied successfully. You are now viewing your copy of the question.',
-                );
-                res.redirect(
-                  res.locals.urlPrefix + '/question/' + result.rows[0].question_id + '/settings',
-                );
-              },
-            );
-          }
+        editor.assertCanEdit();
+        await editor.doEditAsync();
+        res.redirect(req.originalUrl);
+      }
+    } else if (req.body.__action === 'copy_question') {
+      if (idsEqual(req.body.to_course_id, res.locals.course.id)) {
+        // In this case, we are making a duplicate of this question in the same course
+        const editor = new QuestionCopyEditor({
+          locals: res.locals,
         });
-      });
-    } else {
-      callbackify(copyQuestionBetweenCourses)(
-        res,
-        {
+        editor.assertCanEdit();
+        await editor.doEditAsync();
+        const questionId = await sqldb.queryRow(
+          sql.select_question_id_from_uuid,
+          { uuid: editor.uuid, course_id: res.locals.course.id },
+          IdSchema,
+        );
+        flash(
+          'success',
+          'Question copied successfully. You are now viewing your copy of the question.',
+        );
+        res.redirect(res.locals.urlPrefix + '/question/' + questionId + '/settings');
+      } else {
+        copyQuestionBetweenCourses(res, {
```

Comment:
This is an async function, it should be awaited.

```suggestion
        await copyQuestionBetweenCourses(res, {
```

Reply:
Updated locally in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).